src/core/thread/network_diagnostic.cpp

```diff
@@ -495,7 +495,7 @@ void NetworkDiagnostic::HandleDiagnosticGetQuery(Coap::Message &aMessage, const
         SuccessOrExit(error = message->SetPayloadMarker());
     }
 
-    if (aMessageInfo.GetSockAddr().IsLinkLocal() || aMessageInfo.GetSockAddr().IsLinkLocalMulticast())
+    if (aMessageInfo.GetPeerAddr().IsLinkLocal() || aMessageInfo.GetPeerAddr().IsLinkLocalMulticast())
```

Comment:
The source of the query will not be a link local multicast address

```suggestion
    if (aMessageInfo.GetPeerAddr().IsLinkLocal())
```

Reply:
Yes. Thanks.